### DISCUSSION TITLE
Support Shiki's `langAlias` option in markdown configuration

### DISCUSSION THREAD

**Discussion by Sec-ant** [withastro/roadmap#954] [Proposal]
### Body

# Summary

Support Shiki's `langAlias` option in markdown configuration so we can register custom language aliases for the code blocks.

# Background & Motivation

In our website we have some code blocks that are tagged with `cjs`, because they contain code that has `cjs`-specific syntax like `"use strict"`, `module.exports`, .etc. However Shiki doesn't support syntax highlighting for `cjs` blocks out of box and will fallback to use `txt`:

- https://deploy-preview-610--biomejs.netlify.app/linter/rules/no-redundant-use-strict/
- https://app.netlify.com/sites/biomejs/deploys/666c5e50c6f7750008bb8102#L143-L156

Astro provides the [`langs`](https://docs.astro.build/en/guides/markdown-content/#shiki-configuration) option to register custom languages, but that also means we have to double load the `js` grammar for the `cjs` language like we did in https://github.com/biomejs/website/pull/609 which is not very elegant.

On the other hand, there is a [`langAlias`](https://shiki.matsu.io/guide/load-lang#custom-language-aliases) option in Shiki that allows registering aliases for a language, but it is not exposed from Astro.

# Goals

Support `langAlias` option in markdown configuration.

# Example

```mjs
import { defineConfig } from 'astro/config';

export default defineConfig({
  markdown: {
    shikiConfig: {
      langAlias: { 
        cjs: "javascript",
      },
    },
  },
});
```

**Comment by bluwy** [comment]
Seems like an ok addition to me 👍

**Comment by ematipico** [comment]
This will be shipped in Astro v4.16
